### PR TITLE
added wifi scanning module

### PR DESCRIPTION
**Pull request by abhi211199**
manifest has been changed so as wifi activity opens by default(for testing purposes).

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -4,25 +4,29 @@
     package="com.android.splitpersonality">
 
     <dist:module dist:instant="true" />
-
+    <uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
+    <uses-permission android:name="android.permission.CHANGE_WIFI_STATE"/>
+    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
```
Comment (shreyshrey71) [inline app/src/main/AndroidManifest.xml:10]: Add permission for backgroung location

File: app/src/main/java/com/android/splitpersonality/WiFi/WifiActivity.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        receiver = new WifiReceiver();
+        registerReceiver(receiver, new IntentFilter(manager.SCAN_RESULTS_AVAILABLE_ACTION));
+        manager.startScan();
+    }
+    //display keywords based on DBm strength
+    public String returnlevel(int DBm)
+    {
+        if(DBm >= -50) return DBm+" dBm > Excellent";
+        else if (DBm < -50 && DBm >= -60) return DBm+" dBm > Good";
+        else if (DBm < -60 && DBm >= -70) return DBm+" dBm > Fair";
```
Comment (shreyshrey71) [inline app/src/main/java/com/android/splitpersonality/WiFi/WifiActivity.java:95]: Remove unnecessary conditions

File: app/src/main/java/com/android/splitpersonality/WiFi/WifiActivity.java [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+    private BroadcastReceiver wifiStateReceiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            int wifiStateExtra = intent.getIntExtra(WifiManager.EXTRA_WIFI_STATE,
+                    WifiManager.WIFI_STATE_UNKNOWN);
+
+            switch (wifiStateExtra) {
+                case WifiManager.WIFI_STATE_ENABLED:
+                   Log.e("on","on");
```
Comment (shreyshrey71) [inline app/src/main/java/com/android/splitpersonality/WiFi/WifiActivity.java]: change tag to wifi in log

File: app/src/main/java/com/android/splitpersonality/WiFi/WifiActivity.java [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+                        Toast.makeText(WifiActivity.this,"Please give Location! to app",Toast.LENGTH_SHORT).show();
+
+                    }
+                    else {
+                        scanning();
+                        text.setVisibility(View.GONE);
+                    }
+                    break;
+                case WifiManager.WIFI_STATE_DISABLED:
+                    Log.e("on","off");
```
Comment (shreyshrey71) [inline app/src/main/java/com/android/splitpersonality/WiFi/WifiActivity.java]: same here

**Review by shreyshrey71 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/android/splitpersonality/WiFi/WifiActivity.java (hunk 2 above)
Comment (abhi211199) [reply app/src/main/java/com/android/splitpersonality/WiFi/WifiActivity.java:95]: This condition is required so as to show the strength of wifi to the user.

**Pull request closed without merging**